tests/agent_server/test_facts.py

```diff
@@ -431,4 +433,4 @@ async def test_get_fact_no_code(resource_container, client, clienthelper, enviro
     log_entry = result["logs"][0]
     assert log_entry["action"] == "getfact"
     assert log_entry["status"] == "unavailable"
-    assert "Failed to load" in log_entry["messages"][0]["msg"]
+    assert "Unable to deserialize" in log_entry["messages"][0]["msg"]
```

Comment:
For this change and the one in test_server_agent, I don't fully understand what we're testing.
These changes make the tests pass but I'm not sure if we're still testing the same thing as before